### PR TITLE
Regression issue: Allow SSM parameter overwrite upon creation

### LINKED ISSUE #25335: aws_ssm_parameter not allowing overwrite = true
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v1.2.2
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v4.18.0

### Affected Resource(s)

* aws_ssm_parameter

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
terraform {
  required_providers {
    aws = {
      source = "hashicorp/aws"
      version = "~> 4.18.0"
    }
  }
}

provider "aws" {
  region = "us-east-1"
  profile = "develop"
}

variable "environment" {
  default = "develop"
}

resource "aws_ssm_parameter" "environment" {
  name = "/Global/Env"
  description = "Storage of environment variable (develop,staging,prod)"
  type = "String"
  value = var.environment
  overwrite = true
}
```

### Expected Behavior

Overwrite any previously existing parameter store value

### Actual Behavior

```shell
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_ssm_parameter.environment will be created
  + resource "aws_ssm_parameter" "environment" {
      + arn         = (known after apply)
      + data_type   = (known after apply)
      + description = "Storage of environment variable (develop,staging,prod)"
      + id          = (known after apply)
      + key_id      = (known after apply)
      + name        = "/Global/Env"
      + overwrite   = true
      + tags_all    = (known after apply)
      + tier        = (known after apply)
      + type        = "String"
      + value       = (sensitive value)
      + version     = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_ssm_parameter.environment: Creating...
╷
│ Error: error creating SSM Parameter (/Global/Env): ParameterAlreadyExists: The parameter already exists. To overwrite this value, set the overwrite option in the request to true.
│ 
│   with aws_ssm_parameter.environment,
│   on main.tf line 19, in resource "aws_ssm_parameter" "environment":
│   19: resource "aws_ssm_parameter" "environment" {
│ 
╵
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Only observed the incorrect behaviour while running aws provider v4.18.0, rolling back to v4.17.0 overwrite = true seems to function normally.  Once the build has overwritten in v4.17.0, switching back to v4.18.0 obviously no longer needs the overwrite value as its managing properly.  

### References

Nothing I could find for the current release.

### PR DESCRIPTION
**Pull request by bschaatsbergen**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #25335

### Regression bug

This issue was flagged after releasing v4.18.0 through #25335
I've reverted the commit.

Output from acceptance testing:

```
make testacc TESTS=TestAccSSMParameter_ PKG=ssm
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ssm/... -v -count 1 -parallel 20 -run='TestAccSSMParameter_'  -timeout 180m
=== RUN   TestAccSSMParameter_basic
=== PAUSE TestAccSSMParameter_basic
=== RUN   TestAccSSMParameter_tier
=== PAUSE TestAccSSMParameter_tier
=== RUN   TestAccSSMParameter_Tier_intelligentTieringToStandard
=== PAUSE TestAccSSMParameter_Tier_intelligentTieringToStandard
=== RUN   TestAccSSMParameter_Tier_intelligentTieringToAdvanced
=== PAUSE TestAccSSMParameter_Tier_intelligentTieringToAdvanced
=== RUN   TestAccSSMParameter_Tier_intelligentTieringOnCreation
=== PAUSE TestAccSSMParameter_Tier_intelligentTieringOnCreation
=== RUN   TestAccSSMParameter_Tier_intelligentTieringOnUpdate
=== PAUSE TestAccSSMParameter_Tier_intelligentTieringOnUpdate
=== RUN   TestAccSSMParameter_disappears
=== PAUSE TestAccSSMParameter_disappears
=== RUN   TestAccSSMParameter_Overwrite_basic
=== PAUSE TestAccSSMParameter_Overwrite_basic
=== RUN   TestAccSSMParameter_Overwrite_cascade
=== PAUSE TestAccSSMParameter_Overwrite_cascade
=== RUN   TestAccSSMParameter_Overwrite_tags
=== PAUSE TestAccSSMParameter_Overwrite_tags
=== RUN   TestAccSSMParameter_Overwrite_noOverwriteTags
=== PAUSE TestAccSSMParameter_Overwrite_noOverwriteTags
=== RUN   TestAccSSMParameter_Overwrite_updateToTags
=== PAUSE TestAccSSMParameter_Overwrite_updateToTags
=== RUN   TestAccSSMParameter_tags
=== PAUSE TestAccSSMParameter_tags
=== RUN   TestAccSSMParameter_updateType
=== PAUSE TestAccSSMParameter_updateType
=== RUN   TestAccSSMParameter_Overwrite_updateDescription
=== PAUSE TestAccSSMParameter_Overwrite_updateDescription
=== RUN   TestAccSSMParameter_changeNameForcesNew
=== PAUSE TestAccSSMParameter_changeNameForcesNew
=== RUN   TestAccSSMParameter_fullPath
=== PAUSE TestAccSSMParameter_fullPath
=== RUN   TestAccSSMParameter_Secure_basic
=== PAUSE TestAccSSMParameter_Secure_basic
=== RUN   TestAccSSMParameter_Secure_insecure
=== PAUSE TestAccSSMParameter_Secure_insecure
=== RUN   TestAccSSMParameter_Secure_insecureChangeSecure
=== PAUSE TestAccSSMParameter_Secure_insecureChangeSecure
=== RUN   TestAccSSMParameter_DataType_ec2Image
=== PAUSE TestAccSSMParameter_DataType_ec2Image
=== RUN   TestAccSSMParameter_Secure_key
=== PAUSE TestAccSSMParameter_Secure_key
=== RUN   TestAccSSMParameter_Secure_keyUpdate
=== PAUSE TestAccSSMParameter_Secure_keyUpdate
=== CONT  TestAccSSMParameter_basic
=== CONT  TestAccSSMParameter_Secure_key
=== CONT  TestAccSSMParameter_Overwrite_updateToTags
=== CONT  TestAccSSMParameter_DataType_ec2Image
=== CONT  TestAccSSMParameter_Secure_insecureChangeSecure
=== CONT  TestAccSSMParameter_Secure_insecure
=== CONT  TestAccSSMParameter_Secure_basic
=== CONT  TestAccSSMParameter_fullPath
=== CONT  TestAccSSMParameter_changeNameForcesNew
=== CONT  TestAccSSMParameter_Overwrite_updateDescription
=== CONT  TestAccSSMParameter_updateType
=== CONT  TestAccSSMParameter_tags
=== CONT  TestAccSSMParameter_Tier_intelligentTieringToAdvanced
=== CONT  TestAccSSMParameter_Tier_intelligentTieringOnUpdate
=== CONT  TestAccSSMParameter_disappears
=== CONT  TestAccSSMParameter_Overwrite_noOverwriteTags
=== CONT  TestAccSSMParameter_Tier_intelligentTieringOnCreation
=== CONT  TestAccSSMParameter_Overwrite_tags
=== CONT  TestAccSSMParameter_Overwrite_cascade
=== CONT  TestAccSSMParameter_Secure_keyUpdate
--- PASS: TestAccSSMParameter_disappears (25.36s)
=== CONT  TestAccSSMParameter_Overwrite_basic
--- PASS: TestAccSSMParameter_Tier_intelligentTieringOnCreation (31.98s)
=== CONT  TestAccSSMParameter_tier
--- PASS: TestAccSSMParameter_fullPath (34.79s)
=== CONT  TestAccSSMParameter_Tier_intelligentTieringToStandard
--- PASS: TestAccSSMParameter_Overwrite_noOverwriteTags (34.97s)
--- PASS: TestAccSSMParameter_basic (35.38s)
--- PASS: TestAccSSMParameter_Secure_basic (35.51s)
--- PASS: TestAccSSMParameter_Overwrite_tags (35.63s)
--- PASS: TestAccSSMParameter_DataType_ec2Image (39.17s)
--- PASS: TestAccSSMParameter_Secure_key (39.74s)
--- PASS: TestAccSSMParameter_Tier_intelligentTieringOnUpdate (49.40s)
--- PASS: TestAccSSMParameter_Overwrite_updateToTags (51.72s)
--- PASS: TestAccSSMParameter_Overwrite_updateDescription (52.74s)
--- PASS: TestAccSSMParameter_changeNameForcesNew (55.19s)
--- PASS: TestAccSSMParameter_updateType (55.39s)
--- PASS: TestAccSSMParameter_Secure_keyUpdate (59.04s)
--- PASS: TestAccSSMParameter_Secure_insecureChangeSecure (65.56s)
--- PASS: TestAccSSMParameter_Overwrite_cascade (66.96s)
--- PASS: TestAccSSMParameter_tags (70.65s)
--- PASS: TestAccSSMParameter_Secure_insecure (70.85s)
--- PASS: TestAccSSMParameter_Overwrite_basic (45.86s)
--- PASS: TestAccSSMParameter_Tier_intelligentTieringToAdvanced (75.03s)
--- PASS: TestAccSSMParameter_Tier_intelligentTieringToStandard (60.98s)
--- PASS: TestAccSSMParameter_tier (63.99s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/ssm        96.063s
```

### CODE REVIEW

File: internal/service/ssm/parameter_test.go [hunk 1]
```diff
@@ -285,10 +285,31 @@ func TestAccSSMParameter_Overwrite_basic(t *testing.T) {
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
 		CheckDestroy:             testAccCheckParameterDestroy,
 		Steps: []resource.TestStep{
+			{
+
+				PreConfig: func() {
+					conn := acctest.Provider.Meta().(*conns.AWSClient).SSMConn
+
+					input := &ssm.PutParameterInput{
```
Comment (bschaatsbergen) [inline internal/service/ssm/parameter_test.go:293]: "test_parameter" here is tightly coupled to the parameter name set through the [testAccParameterConfig_basicOverwrite](https://github.com/bschaatsbergen/terraform-provider-aws/blob/795c5a862a8e1a72bb468a098c8b74d72293dc42/internal/service/ssm/parameter_test.go#L1000), we could optionally remove the `test_parameter` prefix from the config and just set the `name`.

File: website/docs/r/ssm_parameter.html.markdown [hunk 2]
```diff
@@ -10,6 +10,8 @@ description: |-
 
 Provides an SSM Parameter resource.
 
+~> **Note:** `overwrite` also makes it possible to overwrite an existing SSM Parameter that's not created by Terraform before.
```
Comment (bschaatsbergen) [inline website/docs/r/ssm_parameter.html.markdown:13]: `created by Terraform` or `created through Terraform` ? I'm not quite sure.

**Review by andresvia — APPROVED** [review]

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccSSMParameter_' PKG=ssm ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ssm/... -v -count 1 -parallel 2  -run=TestAccSSMParameter_ -timeout 180m
=== RUN   TestAccSSMParameter_basic
=== PAUSE TestAccSSMParameter_basic
=== RUN   TestAccSSMParameter_tier
=== PAUSE TestAccSSMParameter_tier
=== RUN   TestAccSSMParameter_Tier_intelligentTieringToStandard
=== PAUSE TestAccSSMParameter_Tier_intelligentTieringToStandard
=== RUN   TestAccSSMParameter_Tier_intelligentTieringToAdvanced
=== PAUSE TestAccSSMParameter_Tier_intelligentTieringToAdvanced
=== RUN   TestAccSSMParameter_Tier_intelligentTieringOnCreation
=== PAUSE TestAccSSMParameter_Tier_intelligentTieringOnCreation
=== RUN   TestAccSSMParameter_Tier_intelligentTieringOnUpdate
=== PAUSE TestAccSSMParameter_Tier_intelligentTieringOnUpdate
=== RUN   TestAccSSMParameter_disappears
=== PAUSE TestAccSSMParameter_disappears
=== RUN   TestAccSSMParameter_Overwrite_basic
=== PAUSE TestAccSSMParameter_Overwrite_basic
=== RUN   TestAccSSMParameter_Overwrite_cascade
=== PAUSE TestAccSSMParameter_Overwrite_cascade
=== RUN   TestAccSSMParameter_Overwrite_tags
=== PAUSE TestAccSSMParameter_Overwrite_tags
=== RUN   TestAccSSMParameter_Overwrite_noOverwriteTags
=== PAUSE TestAccSSMParameter_Overwrite_noOverwriteTags
=== RUN   TestAccSSMParameter_Overwrite_updateToTags
=== PAUSE TestAccSSMParameter_Overwrite_updateToTags
=== RUN   TestAccSSMParameter_tags
=== PAUSE TestAccSSMParameter_tags
=== RUN   TestAccSSMParameter_updateType
=== PAUSE TestAccSSMParameter_updateType
=== RUN   TestAccSSMParameter_Overwrite_updateDescription
=== PAUSE TestAccSSMParameter_Overwrite_updateDescription
=== RUN   TestAccSSMParameter_changeNameForcesNew
=== PAUSE TestAccSSMParameter_changeNameForcesNew
=== RUN   TestAccSSMParameter_fullPath
=== PAUSE TestAccSSMParameter_fullPath
=== RUN   TestAccSSMParameter_Secure_basic
=== PAUSE TestAccSSMParameter_Secure_basic
=== RUN   TestAccSSMParameter_Secure_insecure
=== PAUSE TestAccSSMParameter_Secure_insecure
=== RUN   TestAccSSMParameter_Secure_insecureChangeSecure
=== PAUSE TestAccSSMParameter_Secure_insecureChangeSecure
=== RUN   TestAccSSMParameter_DataType_ec2Image
=== PAUSE TestAccSSMParameter_DataType_ec2Image
=== RUN   TestAccSSMParameter_Secure_key
=== PAUSE TestAccSSMParameter_Secure_key
=== RUN   TestAccSSMParameter_Secure_keyUpdate
=== PAUSE TestAccSSMParameter_Secure_keyUpdate
=== CONT  TestAccSSMParameter_basic
=== CONT  TestAccSSMParameter_tags
--- PASS: TestAccSSMParameter_basic (20.53s)
=== CONT  TestAccSSMParameter_disappears
--- PASS: TestAccSSMParameter_disappears (14.56s)
=== CONT  TestAccSSMParameter_Overwrite_updateToTags
--- PASS: TestAccSSMParameter_tags (47.99s)
=== CONT  TestAccSSMParameter_Overwrite_noOverwriteTags
--- PASS: TestAccSSMParameter_Overwrite_updateToTags (30.81s)
=== CONT  TestAccSSMParameter_Overwrite_tags
--- PASS: TestAccSSMParameter_Overwrite_noOverwriteTags (18.02s)
=== CONT  TestAccSSMParameter_Overwrite_cascade
--- PASS: TestAccSSMParameter_Overwrite_tags (24.44s)
=== CONT  TestAccSSMParameter_Overwrite_basic
--- PASS: TestAccSSMParameter_Overwrite_cascade (50.68s)
=== CONT  TestAccSSMParameter_Tier_intelligentTieringToAdvanced
--- PASS: TestAccSSMParameter_Overwrite_basic (63.35s)
=== CONT  TestAccSSMParameter_Tier_intelligentTieringOnUpdate
--- PASS: TestAccSSMParameter_Tier_intelligentTieringToAdvanced (72.38s)
=== CONT  TestAccSSMParameter_Tier_intelligentTieringOnCreation
--- PASS: TestAccSSMParameter_Tier_intelligentTieringOnUpdate (45.79s)
=== CONT  TestAccSSMParameter_Secure_insecure
--- PASS: TestAccSSMParameter_Tier_intelligentTieringOnCreation (30.27s)
=== CONT  TestAccSSMParameter_Secure_keyUpdate
--- PASS: TestAccSSMParameter_Secure_insecure (70.41s)
=== CONT  TestAccSSMParameter_Secure_key
--- PASS: TestAccSSMParameter_Secure_keyUpdate (51.85s)
=== CONT  TestAccSSMParameter_DataType_ec2Image
--- PASS: TestAccSSMParameter_Secure_key (28.96s)
=== CONT  TestAccSSMParameter_Secure_insecureChangeSecure
--- PASS: TestAccSSMParameter_DataType_ec2Image (50.48s)
=== CONT  TestAccSSMParameter_changeNameForcesNew
--- PASS: TestAccSSMParameter_Secure_insecureChangeSecure (65.28s)
=== CONT  TestAccSSMParameter_Secure_basic
--- PASS: TestAccSSMParameter_changeNameForcesNew (49.96s)
=== CONT  TestAccSSMParameter_fullPath
--- PASS: TestAccSSMParameter_Secure_basic (32.82s)
=== CONT  TestAccSSMParameter_Tier_intelligentTieringToStandard
--- PASS: TestAccSSMParameter_fullPath (32.27s)
=== CONT  TestAccSSMParameter_updateType
--- PASS: TestAccSSMParameter_updateType (47.91s)
=== CONT  TestAccSSMParameter_tier
--- PASS: TestAccSSMParameter_Tier_intelligentTieringToStandard (66.43s)
=== CONT  TestAccSSMParameter_Overwrite_updateDescription
--- PASS: TestAccSSMParameter_Overwrite_updateDescription (41.27s)
--- PASS: TestAccSSMParameter_tier (58.31s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ssm	515.061s
```

**Comment by ewbankkit** [comment]
@bschaatsbergen Thanks for the contribution :tada: :clap:.

**Pull request merged**